### DISCUSSION TITLE
Non-Newtonian Fluid Flow in Porous Media

### DISCUSSION THREAD

**Discussion by sinaomrani96** [idaholab/moose#23211] [Q&A General] [answered]
Hi,

I have searched and did not find anything special regarding non-Newtonian fluid flow in porous media. The only conversation was for 7 years ago. So, I was wondering if anybody knows that this has been developed or not. Thanks.

Best regards,
Sina

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont think anything got developed
@WilkAndy for confirmation

Guillaume

**Reply by sinaomrani96** [reply]
Hi, 

Do you have any suggestion regarding developing it on MOOSE? I mean any similar modules such as PorousFlow that can be modified.

Best regards,
Sina

**Reply by WilkAndy** [reply]
I have done non-Newtonian slurry flow with MOOSE, but unfortunately that code cannot be released to the general public.  I did not end up modifying PorousFlow - instead, i made my own App, and my input files used as many Kernels/Materials as possible from PorousFlow (and also TensorMechanics, because i was interested in mechanical feedbacks).   As i remember, the non-Newtonian part wasn't difficult, but i had a moving front, of a non-Newtonian slurry invading another fluid, and that was numerically tricky.   Sorry, i can't help any further.
